File: frontend/src/app/teams/submodules/tasks/models/tasks-filter.dto.ts
export enum TasksViewTypes {
	KANBAN = 'KANBAN',
	LIST = 'LIST',
	TABLE = 'TABLE',
}

export interface TasksFilterDTO {
	searchValue?: string
	view?: string
	executorId?: number
}

export interface TasksExportDTO {
	startDate?: string
	endDate?: string
	executorId?: number
	format?: string
}

File: backend/src/main/java/org/kaiteki/backend/teams/modules/tasks/models/dto/TasksFilterDTO.java
package org.kaiteki.backend.teams.modules.tasks.models.dto;

import lombok.Builder;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;
import java.util.List;
import java.util.Set;

@Data
@Builder
public class TasksFilterDTO {
    private Long statusId;
    private String searchValue;
    private Long executorId;
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
    private LocalDate startDate;
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
    private LocalDate endDate;
    private Long teamId;
    private Set<Long> teamIds;
}
